ensime_shared/protocol.py

```diff
@@ -45,18 +45,20 @@ def register_responses_handlers(self):
 
     def handle_incoming_response(self, call_id, payload):
         """Get a registered handler for a given response and execute it."""
-        self.log("handle_incoming_response: in {}".format(payload))
+        self.log.debug('handle_incoming_response: in %s', payload)
```

Comment:
This one is actually called like 10 lines after the `unqueue` log message in the comment just above that outputs the raw JSON, which is apparently already formatted. So they're usually consecutive log lines and it looks pretty redundant and noisy to me (even for debug logging) to have both. I think we're pretty certain at this stage that there is no bug in our call to `json.loads` 😄 

So what seems sensible to me then is a one-line message here, showing `call_id` instead of the payload, to make sense of things if you're analyzing a few messages arriving concurrently. I guess it can also include `typehint` for server-initiated messages with no call ID.